app/controllers/api/actions_controller.rb

```diff
@@ -11,7 +11,14 @@ def edit_resource(type, id = nil, data = {})
     end
 
     def options
-      render_options(:actions, :action_types => MiqAction::TYPES)
+      render_options(:actions, build_action_options)
+    end
+
+    def build_action_options
```

Comment:
Can you make this a private method?  Public methods are ones expected to be routed to via the API, so this should be private.

Reply:
The other option is just to multiline the hash in the existing method:

```ruby
def options
  render_options(:actions, :action_types => MiqAction::TYPES,
                           :snmp_trap    => MiqSnmp.available_types)
```